### PR TITLE
fix build for lasso/utils.h and various LASSO_SIGNATURE_METHOD_ options

### PR DESCRIPTION
**Pull request by aanderse**
Currently `configure.ac` depends on a few libraries being implicitly found on a system.

Here is an example of `config.log` (debug output from the build system) when compiling `mod_auth_mellon` on my system:
```
configure:4408: checking for lasso/utils.h
configure:4408: gcc -c -g -O2 -I/nix/store/0znvspc0hzgw0gfy0vm3qvfknh8xbcij-lasso-2.7.0/include  -I/nix/store/0znvspc0hzgw0gfy0vm3qvfknh8xbcij-lasso-2.7.0/include conftest.c >&5
In file included from conftest.c:51:
/nix/store/0znvspc0hzgw0gfy0vm3qvfknh8xbcij-lasso-2.7.0/include/lasso/utils.h:28:10: fatal error: glib.h: No such file or directory
   28 | #include <glib.h>
      |          ^~~~~~~~
compilation terminated.
```

**compilation output before this PR**
```
checking for lasso/utils.h... no
checking for gcc options needed to detect all undeclared functions... none needed
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA256 is declared... no
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA384 is declared... no
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA512 is declared... no
```

**compilation output after this PR**
```
checking for lasso/utils.h... yes
checking for gcc options needed to detect all undeclared functions... none needed
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA256 is declared... yes
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA384 is declared... yes
checking whether LASSO_SIGNATURE_METHOD_RSA_SHA512 is declared... yes
```

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -106,15 +106,15 @@ AC_SUBST(XMLSEC_LIBS)
 # found in the LASSO_CFLAGS. Save and restore CFLAGS and CPPFLAGS.
 saved_CFLAGS=$CFLAGS
 saved_CPPFLAGS=$CPPFLAGS
-CFLAGS="$CFLAGS $pkg_cv_LASSO_CFLAGS"
-CPPFLAGS="$CPPFLAGS $pkg_cv_LASSO_CFLAGS"
+CFLAGS="$CFLAGS $pkg_cv_LASSO_CFLAGS $GLIB_CFLAGS $XMLSEC_CFLAGS"
```
Comment (jhrozek) [inline configure.ac:109]: wouldn't it be more consistent to use `$pkg_cv_GLIB_CFLAGS`? It's probably the same values, it would just look more similar..

**Review by jhrozek — COMMENTED** [review]
sorry nobody checked the PR earlier. One nit inline.

File: configure.ac (hunk 1 above)
Comment (aanderse) [reply configure.ac:109]: @jhrozek thanks for following up. I have to confess I know almost nothing about this build system. I'll go ahead and replace all my instances of `$GLIB_CFLAGS` with `$pkg_cv_GLIB_CFLAGS` then? Same for `$XMLSEC_CFLAGS` -> `$pkg_cv_XMLSEC_CFLAGS`?

Comment (jhrozek) [reply configure.ac:109]: you know I was probably being over pedantic. Let's get the patch merged, I don't see anything wrong about it tbh. If others do, they had ~20 days to review the PR and can always send another PR on top :-P

**Comment by aanderse** [comment]
Thank you very much @jhrozek!

**Pull request merged**